ion/integration_test.go

```diff
@@ -529,13 +529,43 @@ func writeToWriterFromReader(t *testing.T, reader Reader, writer Writer) {
 			}
 
 		case IntType:
-			val, err := reader.BigIntValue()
+			intSize, err := reader.IntSize()
 			if err != nil {
-				t.Errorf("Something went wrong when reading Int value. " + err.Error())
+				t.Errorf("Something went wrong when retrieving the Int size. " + err.Error())
 			}
-			err = writer.WriteBigInt(val)
-			if err != nil {
-				t.Errorf("Something went wrong when writing Int value. " + err.Error())
+
+			switch intSize {
+			case Int32, Int64:
+				val, err := reader.Int64Value()
+				if err != nil {
+					t.Errorf("Something went wrong when reading Int value. " + err.Error())
+				}
+
+				err = writer.WriteInt(val)
+				if err != nil {
+					t.Errorf("Something went wrong when writing Int value. " + err.Error())
+				}
+			case Uint64:
```

Comment:
I realize this is out of scope for this PR, but I was a bit surprised to see the mix of signed/unsigned `IntSize` variants. I've [opened an issue](https://github.com/amzn/ion-go/issues/102) to discuss whether we'd like to keep this API in a 1.0 release.